packages/node/src/integrations/tracing/express.ts

```diff
@@ -18,6 +18,15 @@ const _expressIntegration = (() => {
             requestHook(span) {
               addOriginToSpan(span, 'auto.http.otel.express');
             },
+            spanNameHook(info, defaultName) {
+              if (info.layerType === 'request_handler') {
+                // type cast b/c Otel unfortunately types info.request as any :(
+                const req = info.request as { method?: string };
+                const method = req.method ? req.method.toUpperCase() : 'GET';
+                getIsolationScope().setTransactionName(`${method} ${info.route}`);
```

Comment:
I guess there's also a case to be made that this should be set on the  current scope instead. 
Rationale: 2 nested request handlers -> current scope would set the transactionName correctly whereas isolation scope would outlive the inner handler